entity-service-impl/src/main/java/org/hypertrace/entity/query/service/EntityQueryServiceImpl.java

```diff
@@ -346,6 +350,70 @@ public void bulkUpdate(
     }
   }
 
+  @Override
+  public void bulkUpdateEntityArrayAttribute(
+      BulkEntityArrayAttributeUpdateRequest request,
+      StreamObserver<ResultSetChunk> responseObserver) {
+    RequestContext requestContext = RequestContext.CURRENT.get();
+    String tenantId = requestContext.getTenantId().orElse(null);
+    if (isNull(tenantId)) {
+      responseObserver.onError(new ServiceException("Tenant id is missing in the request."));
+      return;
+    }
+    try {
+      Set<Key> keys =
+          request.getEntityIdsList().stream()
+              .map(entityId -> new SingleValueKey(tenantId, entityId))
+              .collect(Collectors.toSet());
+
+      String attributeId = request.getAttribute().getColumnName();
+
+      String subDocPath =
+          entityAttributeMapping
+              .getDocStorePathByAttributeId(requestContext, attributeId)
+              .orElseThrow(
+                  () -> new IllegalArgumentException("Unknown attribute FQN " + attributeId));
+
+      List<Document> subDocuments = new ArrayList<>();
+      for (LiteralConstant literalConstant : request.getValuesList()) {
+        subDocuments.add(convertToJsonDocument(literalConstant));
+      }
+      BulkArrayValueUpdateRequest bulkArrayValueUpdateRequest =
+          new BulkArrayValueUpdateRequest(
+              keys,
+              subDocPath + ARRAY_VALUE_PATH_SUFFIX,
```

Comment:
How do we handle this elsewhere? Do we write the whole value doc and here we're only doing a subset of the value doc? I wonder if there's a way for us to pull this out of the proto, since this is based on the shape of the `AttributeValueList` message. That said, aren't the subdocuments we're printing here `AttributeValue`s? That seems like a mismatch to me. I'd expect this path to show up in the subdocs too (which would prevent them from merging).

Can we add a test that verifies the `BulkArrayValueUpdateRequest` ?

Reply:
I tried to extract it from AttributeValue -> AttributeValueList. It was just too complicated to extract that path.

AttributeValueList actually contains repeated AttributeValue. 
```
message AttributeValueList {
  repeated AttributeValue values = 1;
}
```

Reply:
Ok, got it - my mistake. My confusion was that I thought for some reason we were converting an `AttributeValue` holding a list, rather than a list of individual `AttributeValue` scalars. The full path would look like
`attributes.<attrName>.valueList.values[value.string]` for a list of strings, and we're passing an array of docs for path `attributes.<attrName>.valueList.values`, those docs each representing a `value.string` so all good.


As for extracting the path, it's a bit more verbose, but shouldn't be particularly complicated:
```java
Stream.of(
        AttributeValue.getDescriptor().findFieldByNumber(VALUE_LIST_FIELD_NUMBER).getJsonName(),
        AttributeValueList.getDescriptor().findFieldByNumber(VALUES_FIELD_NUMBER).getJsonName()
).collect(joining("."))
```